Patch by Robert Scheck <robert@fedoraproject.org> for clamav <= 0.98 which adds support for
ARM. The syscall is taken from arch/arm/include/asm/unistd.h from the Linux kernel. Without
this patch building on (at least) ARMv7 fails like this:

In file included from fan.c:38:0:
fan-syscalllib.h:14:3: error: #error "System call numbers not defined for this architecture"
 # error "System call numbers not defined for this architecture"
   ^
fan-syscalllib.h: In function 'fanotify_init':
fan-syscalllib.h:19:17: error: '__NR_fanotify_init' undeclared (first use in this function)
  return syscall(__NR_fanotify_init, flags, event_f_flags);
                 ^
fan-syscalllib.h:19:17: note: each undeclared identifier is reported only once for each function it appears in
fan-syscalllib.h: In function 'fanotify_mark':
fan-syscalllib.h:25:17: error: '__NR_fanotify_mark' undeclared (first use in this function)
  return syscall(__NR_fanotify_mark, fanotify_fd, flags, mask,
                 ^

--- clamav-0.98/clamd/fan-syscalllib.h			2013-09-16 21:28:14.000000000 +0200
+++ clamav-0.98/clamd/fan-syscalllib.h.arm-fanotify	2013-10-06 22:29:26.000000000 +0200
@@ -10,6 +10,9 @@
 #elif defined(__i386__)
 # define __NR_fanotify_init	338
 # define __NR_fanotify_mark	339
+#elif defined(__ARM_EABI__)
+# define __NR_fanotify_init	367
+# define __NR_fanotify_mark	368
 #else
 # error "System call numbers not defined for this architecture"
 #endif
